public/app/features/logs/utils.ts

```diff
@@ -220,18 +220,20 @@ export const mergeLogsVolumeDataFrames = (dataFrames: DataFrame[]): { dataFrames
 
     configs[level] = {
       meta: dataFrame.meta,
-      valueFieldConfig: valueField.config,
-      timeFieldConfig: timeField.config,
+      valueFieldConfig: valueField?.config ?? {},
+      timeFieldConfig: timeField?.config ?? {},
     };
 
-    for (let pointIndex = 0; pointIndex < length; pointIndex++) {
-      const time: number = timeField.values[pointIndex];
-      const value: number = valueField.values[pointIndex];
-      aggregated[level] ??= {};
-      aggregated[level][time] = (aggregated[level][time] || 0) + value;
+    if (timeField && valueField) {
+      for (let pointIndex = 0; pointIndex < length; pointIndex++) {
+        const time: number = timeField.values[pointIndex];
+        const value: number = valueField.values[pointIndex];
+        aggregated[level] ??= {};
+        aggregated[level][time] = (aggregated[level][time] || 0) + value;
 
-      totals[time] = (totals[time] || 0) + value;
-      maximumValue = Math.max(totals[time], maximumValue);
+        totals[time] = (totals[time] || 0) + value;
+        maximumValue = Math.max(totals[time], maximumValue);
```

Comment:
not sure if it matters here, but doing an `if` check is usually faster than always mutating `maximumValue`:

```js
if (maximumValue < totals[time])
  maximumValue = totals[time];
```

Reply:
Of course it matters, thanks for the suggestion!